realm/realm-library/src/androidTest/kotlin/io/realm/UnmanagedSetTester.kt

```diff
@@ -0,0 +1,296 @@
+                        notPresentValue = VALUE_NUMERIC_NOT_PRESENT.toFloat()
+                )
+            SetSupportedType.DOUBLE ->
+                UnmanagedSetTester<Double>(
+                        testerName = "Double",
+                        values = listOf(VALUE_NUMERIC_HELLO.toDouble(), VALUE_NUMERIC_BYE.toDouble(), null),
+                        notPresentValue = VALUE_NUMERIC_NOT_PRESENT.toDouble()
+                )
+            SetSupportedType.STRING ->
+                UnmanagedSetTester<String>(
+                        testerName = "String",
+                        values = listOf(VALUE_STRING_HELLO, VALUE_STRING_BYE, null),
+                        notPresentValue = VALUE_STRING_NOT_PRESENT
+                )
+            SetSupportedType.BOOLEAN ->
+                UnmanagedSetTester<Boolean>(
+                        testerName = "Boolean",
+                        values = listOf(VALUE_BOOLEAN_HELLO, null),
+                        notPresentValue = VALUE_BOOLEAN_NOT_PRESENT
+                )
+            SetSupportedType.DATE ->
+                UnmanagedSetTester<Date>(
+                        testerName = "Date",
+                        values = listOf(VALUE_DATE_HELLO, VALUE_DATE_BYE, null),
+                        notPresentValue = VALUE_DATE_NOT_PRESENT
+                )
+            SetSupportedType.DECIMAL128 ->
+                UnmanagedSetTester<Decimal128>(
+                        testerName = "Decimal128",
+                        values = listOf(VALUE_DECIMAL128_HELLO, VALUE_DECIMAL128_BYE, null),
+                        notPresentValue = VALUE_DECIMAL128_NOT_PRESENT
+                )
+            SetSupportedType.BINARY ->
+                UnmanagedSetTester<ByteArray>(
+                        testerName = "ByteArray",
+                        values = listOf(VALUE_BINARY_HELLO, VALUE_BINARY_BYE, null),
+                        notPresentValue = VALUE_BINARY_NOT_PRESENT
+                )
+            SetSupportedType.OBJECT_ID ->
+                UnmanagedSetTester<ObjectId>(
+                        testerName = "ObjectId",
+                        values = listOf(VALUE_OBJECT_ID_HELLO, VALUE_OBJECT_ID_BYE, null),
+                        notPresentValue = VALUE_OBJECT_ID_NOT_PRESENT
+                )
+            SetSupportedType.UUID ->
+                UnmanagedSetTester<UUID>(
+                        testerName = "UUID",
+                        values = listOf(VALUE_UUID_HELLO, VALUE_UUID_BYE, null),
+                        notPresentValue = VALUE_UUID_NOT_PRESENT
+                )
+            SetSupportedType.LINK ->
+                UnmanagedSetTester<RealmModel>(
+                        testerName = "UnmanagedRealmModel",
+                        values = listOf(VALUE_LINK_HELLO, VALUE_LINK_BYE, null),
+                        notPresentValue = VALUE_LINK_NOT_PRESENT
+                )
+            // Ignore Mixed in this switch
```

Comment:
Maybe return null for `MIXES` and throw on `else` to cause the tests to fail if you add new types to the set.

Reply:
Good point, it feels more bulletproof 👍